Enable custom gradients SavedModel option in test

k-w-w opened the pull request:
TensorFlow SavedModel added the option to save custom gradients in [e6c7ba](https://github.com/tensorflow/tensorflow/commit/e6c7ba3875634d7c142d787c2c5012cea9ad8daa). This change updates the JAX SavedModel test to use this option.

k-w-w committed:
Enable custom gradients SavedModel option in test

gnecula commented on jax/experimental/jax2tf/tests/savedmodel_test.py:
```diff
@@ -94,13 +97,8 @@ def f_jax_jvp(primals, tangents):
-    with self.assertRaisesRegex(TypeError, "An op outside of the function building code is being passed"):
-      _ = tape.gradient(y, xv)
+    self.assertAllClose(tape.gradient(y, xv).numpy(),
+                        jax.grad(f_jax)(0.7).astype(np.float32))
```
We should use here `x` in lieu of `0.7`.

k-w-w replied:
Updated the `x` and documentation

gnecula reviewed (commented):
Thank you so much for fixing the custom_gradient support. It has been a long-standing issue for us. 

I would recommend revising also the text in https://github.com/google/jax/blob/3ec5bc503139985c8d099f2ba8bdb71fc059d051/jax/experimental/jax2tf/README.md#no-savedmodel-fine-tuning. I would say instead that we support saving only for the first-order gradient computation.

k-w-w committed:
address comments and update documentation

gnecula reviewed (approved):
LGTM. Thanks!

gnecula committed:
Update savedmodel_test.py

I added a cast to `float32`. This is needed because of an obscure bug in JAX (#6874).

gnecula committed:
Update savedmodel_test.py

Minor change, just to trigger a copybara re-import.
